deploy/operator/ztp/deploy_sno.sh

```diff
@@ -0,0 +1,89 @@
+export ASSISTED_PRIVATEKEY_NAME="${ASSISTED_PRIVATEKEY_NAME:-assisted-ssh-private-key}"
+export EXTRA_BAREMETALHOSTS_FILE="${EXTRA_BAREMETALHOSTS_FILE:-/home/test/dev-scripts/ocp/ostest/extra_baremetalhosts.json}"
+# TODO: make SSH public key configurable
+
+echo "Step configuration:"
+env | sort
+
+echo "Enabling configuration of BMH resources outside of openshift-machine-api namespace"
+oc patch provisioning provisioning-configuration --type merge -p '{"spec":{"watchAllNamespaces": true}}'
+
+echo "Extract information about extra worker nodes..."
+jq --raw-output '.[] | .name + " " + .ports[0].address + " " + .driver_info.username + " " + .driver_info.password + " " + .driver_info.address' "${EXTRA_BAREMETALHOSTS_FILE}" \
+    | while read BMH_NAME MAC_ADDRESS username password ADDRESS ; do
+
+    ENCODED_USERNAME=$(echo -n "${username}" | base64)
+    ENCODED_PASSWORD=$(echo -n "${password}" | base64)
+
+echo "Running Ansible playbook to create kubernetes objects"
+export BMH_NAME MAC_ADDRESS ENCODED_USERNAME ENCODED_PASSWORD ADDRESS
+ansible-playbook assisted-installer-crds-playbook.yaml
+
+echo "Metal3 pods"
+oc get pods -A | grep metal3
+
+echo "VM status"
+virsh list --all
+vm=""
+vm=$(virsh list --all | grep "shut off" | awk '{print $2}')
+echo "Offline VM for ZTP is $vm"
+
+echo "Creating Assisted Installer Pull Secret"
+oc create secret generic ${ASSISTED_PULLSECRET_NAME} --from-file=.dockerconfigjson=${ASSISTED_PULLSECRET_JSON} --type=kubernetes.io/dockerconfigjson -n ${ASSISTED_NAMESPACE}
+
+echo "Creating Assisted Installer Private Key"
+oc create secret generic ${ASSISTED_PRIVATEKEY_NAME} --from-file=ssh-privatekey=/root/.ssh/id_rsa --type=kubernetes.io/ssh-auth -n ${ASSISTED_NAMESPACE}
+
+echo "Creating Assisted Installer Cluster Image Set"
+oc create -f generated/clusterImageSet.yaml
+
+echo "Creating Assisted Installer SNO ClusterDeployment"
+oc create -f generated/clusterDeployment.yaml
+
+echo "Creating Assisted Installer InfraEnv"
+oc create -f generated/infraEnv.yaml
+
+echo "Creating Assisted Installer AgentClusterInstall"
+oc create -f generated/agentClusterInstall.yaml
+
+echo "Creating new baremetal host using baremetal operator"
+oc create -f generate/baremetalHost.yaml
+
+echo "VM status"
+virsh list --all
+oc get pods -A | grep metal3
+
+echo "List clusterdeploments"
+oc get clusterdeployment -n ${ASSISTED_NAMESPACE}
+oc get clusterdeployment -n ${ASSISTED_NAMESPACE} -o json | jq '.status.conditions[] | select(.reason | contains("AgentPlatformState"))'
+
+status=""
+for (( i=0; i<5; i++ ))
+do
+  status=$(virsh list --all | grep $vm | awk '{print $3}')
+  if [ "$status" != "running" ]
+  then
+    echo "$vm is not running ($i retries)"
+    sleep 1m
+  else
+    echo "$vm was started"
+    break
+  fi
+done
+
+if [ "$status" != "running" ]
+then
+  echo "$vm never started"
+  exit 1
+else
+  echo "$vm was started"
+fi
```

Comment:
We don't really care about the VM. we should check if an `agent` CR was created matching the amount of hosts

Reply:
right
fixed